### PR TITLE
upgpatch: aida-x, ver=1.1.0-1

### PR DESCRIPTION
**Pull request by YHStar**
- fix vst3

### CODE REVIEW

File: aida-x/dpf.patch
Comment (wszqkzqk) [inline aida-x/dpf.patch:1]: Did you send it upstream?

Comment (YHStar) [reply aida-x/dpf.patch:1]: Yes, I send to upsteam's dependence at [Add loongarch64 support](https://github.com/DISTRHO/DPF/pull/476)

Comment (wszqkzqk) [reply aida-x/dpf.patch:1]: Why not use http source?

**Comment by YHStar** [comment]
because upstream use commit arch, I can just make pr with upstream’s upstream, then if i want to patch, i must copy file to the right directory due to patch head
 闫宏远

18747853966

  
---- Replied Message ----
From wszqkzqk ***@***.***> Date 11/26/2024 10:32 To lcpu-club/loongarch-packages ***@***.***> Cc YanHongyuan ***@***.***>,
Author ***@***.***> Subject Re: [lcpu-club/loongarch-packages] upgpatch: aida-x, ver=1.1.0-1 (PR #305)  

@wszqkzqk commented on this pull request.
On aida-x/dpf.patch:
Why not use http source?

**Comment by wszqkzqk** [comment]
> because upstream use commit arch, I can just make pr with upstream’s upstream, then if i want to patch, i must copy file to the right directory due to patch head
> […](#)
> ---- Replied Message ---- From wszqkzqk ***@***.***> Date 11/26/2024 10:32 To lcpu-club/loongarch-packages ***@***.***> Cc YanHongyuan ***@***.***>, Author ***@***.***> Subject Re: [lcpu-club/loongarch-packages] upgpatch: aida-x, ver=1.1.0-1 (PR #305) @wszqkzqk commented on this pull request. On aida-x/dpf.patch: Why not use http source? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***>

No, you can pass `-d` to `patch`.

Try this:

```diff
diff --git a/PKGBUILD b/PKGBUILD
index dd6a5b8..36a1fd4 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -38,6 +38,11 @@ _pick() {
   done
 }
 
+prepare() {
+  cd AIDA-X-$pkgver
+  patch -p1 -d "modules/dpf" -i "${srcdir}/add-loongarch-support.patch"
+}
+
 build() {
   local cmake_options=(
     -B build
@@ -174,3 +179,7 @@ package_aida-x-vst3() {
   # mv -v $pkgname/* "$pkgdir"
   install -vDm 755 build/bin/$_name.vst3/Contents/$CARCH-linux/*.so -t "$pkgdir/usr/lib/vst3/$_name.vst3/Contents/$CARCH-linux/"
 }
+
+source+=("add-loongarch-support.patch::https://github.com/DISTRHO/DPF/commit/9c11acb038e4c8494519a4e271e046b49bd2eaea.patch")
+sha512sums+=('d94c7a95c0602755e79624511b1862cdb4985986522990274f48373cb261210a2b558030692a290a755ee153799874bbb7cf83edd3754616cb72121d1d4e5ec8')
+b2sums+=('c214508c3d2b759a490cce4b8ce26b66a173425d52fd1ff7b9323f877cff2614acdd46a8be352a6f36c18db1f7bafee6e4466f15cc1134be2ee679379db7ce9c')

```

**Review by wszqkzqk — APPROVED** [review]

**Pull request merged**